### PR TITLE
[447] Display and editing of ab/provider type in support UI

### PR DESCRIPTION
**Pull request by george-bit**
### Context

As  part of ongoing improvements to the Support Console, support users need the abilty to view and edit whether a provider is an accrediting body and their provider type

### Changes proposed in this pull request

Whilst working on this it occurred to me that  provider type has an assigned method on the  Provider model which sets the `accrediting_body` attribute. After checking with policy this is correct behaviour. This means that we only need to implement the ability to update `provider_type` via the form and then display the `accrediting_body` value on the show page.

I've added a component to disply the correct info in the correct format and andded radio buttons to the form.

new provider view:
image

edit screen: 
image

updated record:
image

### Checklist

- [x] Make sure all information from the Trello card is in here
- [x] Attach to Trello card
- [x] Rebased master
- [x] Cleaned commit history
- [x] Tested by running locally

### CODE REVIEW

**Comment by george-bit** [comment]
[Display and editing of AB in support UI](https://trello.com/c/uPhqtdog/447-display-and-editing-of-ab-in-support-ui)

**Comment by george-bit** [comment]
[Display and editing of provider_type in support UI](https://trello.com/c/SmljIH6F/446-display-and-editing-of-providertype-in-support-ui)

**Review by defong — CHANGES_REQUESTED** [review]

File: app/views/support/providers/edit.html.erb [hunk 1]
```diff
@@ -1,10 +1,19 @@
...
   <div class="govuk-grid-column-two-thirds-from-desktop">
     <%= form_with model: [:support, @provider], local: true do |f| %>
-        
+
     <%= f.govuk_error_summary %>
 
-    <%= f.govuk_text_field :provider_name, label: { text: "New Provider name", size: "s" }, width: 20 %>
+    <%= f.govuk_text_field :provider_name, label: { text: "Provider name", size: "m" }, width: 20 %>
+
+      <%= f.govuk_radio_buttons_fieldset :provider_type, legend: { text: "Provider type", size: "m" }, hint: { text: "Choosing SCITT or University will also mark Provider as an Accredited body" } do %>
```
[the remark this answers is not in the document]
Comment (george-bit) [reply app/views/support/providers/edit.html.erb:11]: Hey, do you mean remove the hon text and add something like:
```suggestion
      <%= f.govuk_radio_button :provider_type, :scitt, label: { text: "SCITT (Accredited body)", size: "s" }, width: 20 %>
        <%= f.govuk_radio_button :provider_type, :lead_school, label: { text: "Lead school (Not an accredited body)", size: "s" }, width: 20 %>
        <%= f.govuk_radio_button :provider_type, :university, label: { text: "University (Accredited body)", size: "s" }, width: 20 %>
```

File: app/views/support/providers/edit.html.erb [hunk 2]
```diff
@@ -1,10 +1,20 @@
...
+
     <%= f.govuk_error_summary %>
 
-    <%= f.govuk_text_field :provider_name, label: { text: "New Provider name", size: "s" }, width: 20 %>
+    <%= f.govuk_text_field :provider_name, label: { text: "Provider name", size: "m" }, width: 20 %>
+
+      <%= f.govuk_radio_buttons_fieldset :provider_type, legend: { text: "Provider type", size: "m" } do %>
+        <%= f.govuk_radio_button :provider_type, :scitt, label: { text: "SCITT", size: "s" }, hint: { text: "Marks as an accredited provider" }, width: 20 %>
+        <%= f.govuk_radio_button :provider_type, :lead_school, label: { text: "Lead school", size: "s" }, hint: { text: "Marks as not an accredited provider" }, width: 20 %>
+        <div class="govuk-radios__divider">or</div>
```
Comment (defong) [inline app/views/support/providers/edit.html.erb:14]:
```suggestion
<%= f.govuk_radio_divider %>
```

**Review by defong — APPROVED** [review]
Just a minor thing

**Review by mbocc — APPROVED** [review]
All approved - If I click on Uni or SCITT, the Accredited Body field automatically changes to YES. If I click on Lead School, it changes to NO. 
Great work, thank you!

**Pull request merged**